tests/shared/FlowManagerSettingsPutTestDefs.js

```diff
@@ -0,0 +1,465 @@
+                    "gpiiKey": gpii.tests.cloud.oauth2.settingsPut.updateSnapset.carlaKey
+                },
+                headers: {
+                    "Authorization": "Bearer token" // set at test run
+                },
+                method: "PUT",
+                expectedStatusCode: 404,
+                expectedErrorMessageStart:
+                    "Can't update preferences for readOnly GPII key \"" +
+                    gpii.tests.cloud.oauth2.settingsPut.updateSnapset.carlaKey +
+                    "\" (perhaps it is a snapset) while executing HTTP PUT"
+            }
+        }
+    }
+});
+
+fluid.defaults("gpii.tests.cloud.oauth2.settingsPut.disruption.updateSnapsetFailure", {
+    gradeNames: ["gpii.test.disruption.settings.sequenceGrade"],
+    testCaseGradeNames: "gpii.tests.cloud.oauth2.settingsPut.updateSnapsetRequests",
+    sequenceElements: {
+        updateSnapsetSequence: {
+            priority: "after:startServer",
+            gradeNames: "gpii.tests.cloud.oauth2.settingsPut.updateSnapsetSequence"
+        }
+    }
+});
+
+// Tests showing the inability to update a snapset
+gpii.tests.cloud.oauth2.settingsPut.updateSnapsetTest = [{
+    testDef: {
+        name: "Flow manager tests - attempt (and fail) to update a snapset"
+    },
+    disruptions: [{
+        sequenceGrade: "gpii.tests.cloud.oauth2.settingsPut.disruption.updateSnapsetFailure"
+    }]
+}];
+
+gpii.tests.cloud.oauth2.settingsPut.updateSnapset.testStatusCode = function (data, request) {
+    jqUnit.assertEquals(
+        "Checking status of " + request.options.path,
+        request.options.expectedStatusCode, request.nativeResponse.statusCode
+    );
+};
+
+gpii.tests.cloud.oauth2.settingsPut.updateSnapset.testResponse = function (data, request) {
+    gpii.tests.cloud.oauth2.settingsPut.updateSnapset.testStatusCode(data, request);
+    var actual = JSON.parse(data);
+
+    jqUnit.assertTrue(
+        "Checking payload error for " + request.options.path,
+        actual.isError
+    );
+    jqUnit.assertTrue(
+        "Checking payload error message for " + request.options.path,
+        actual.message.startsWith(request.options.expectedErrorMessageStart)
+    );
+};
+
+gpii.tests.cloud.oauth2.settingsPut.updateSnapset.testAccessResponse = function (data, request) {
+    var token = JSON.parse(data);
+    var auth = "Bearer " + token.access_token;
+    request.options.stashedAuth = auth;
+
+    gpii.tests.cloud.oauth2.settingsPut.updateSnapset.testStatusCode(data, request);
+    jqUnit.assertNotNull("Checking 'access_token'", token.access_token);
+    jqUnit.assertNotNull("Checking 'expiresIn'", token.expiresIn);
+    jqUnit.assertEquals("Checking 'token_type'",  "Bearer", token.token_type);
+};
+
+gpii.tests.cloud.oauth2.settingsPut.updateSnapset.testLifecycleResponse = function (data, request) {
+    gpii.tests.cloud.oauth2.settingsPut.updateSnapset.testStatusCode(data, request);
+
+    var lifeCycle = JSON.parse(data);
```

Comment:
Could you normalise capitalisation of "lifeCycle" to "lifecycle" throughout, consistent with usage in the core. Note that this function itself already uses "Lifecycle" in its name. Thanks!

Reply:
Okay